### ISSUE TITLE
Cannot specify 'retain_graph' for tensor::backward()

### ISSUE THREAD

**Issue by joverwey** [LaurentMazare/tch-rs#561]
I am training a WGAN which requires the discriminator/critic to be called multiple times before updating the generator.
In order to call `loss.backward()` multiple time, in PyTorch, one has to specify that the graph should be retained using `retain_graph`, otherwise it is deallocated after the first backward() pass.

This is not supported in `tch` even though libtorch does support it. Note that `keep_graph` and `create_graph` below are set to 0.

```
    pub fn f_backward(&self) -> Result<(), TchError> {
        unsafe_torch_err!(at_backward(self.c_tensor, 0, 0));
        Ok(())
    }
```
